src/seedcase_sprout/core/sprout_checks/constants.py

```diff
@@ -0,0 +1,11 @@
+"""Constants for the sprout_checks module."""
+
+"""regex for UUID4, from https://gist.github.com/johnelliott/cf77003f72f889abbc3f32785fa3df8d?permalink_comment_id=4318295#gistcomment-4318295"""
+UUID4_PATTERN = r"[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}"
+
+"""regex for timestamp with format '%Y-%m-%dT%H%M%SZ'. Should match the format used by
+BATCH_TIMESTAMP_FORMAT in sprout.core"""
+TIMESTAMP_PATTERN = r"^\d{4}-\d{2}-\d{2}T\d{6}Z"
```

Comment:
```suggestion
TIMESTAMP_PATTERN = r"^\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])T([01][0-9]|2[0-3])[0-5][0-9][0-5][0-9]Z$"
```
This will match any 4 digit year, but only possible months (01-12), possible dates (01-31) and possible times (hour: 01-23, minutes: 00-59, seconds: 00-59). We could go stricter on the year as well.

Reply:
How about combining your original approach and your new one?
So in one check something like:
- Check that the first 18 chars are a real datetime using `strptime`. This way you don't have to write a regex that can tell apart real datetimes from nonexistent ones (e.g. 2024-02-31T050000Z looks real but doesn't exist).
- Check the rest of the filename. This bit depends on how strict we want to be (as you wrote above). E.g. check that the rest is a uuid.

Reply:
@martonvago Is this to avoid the regex being too complex, or? 

Maybe a nice middle group could be to: 
1. Keep the approach of not expecting the datetime to be in a specific position (so use a rather simple regex like `r"^\d{4}-\d{2}-\d{2}T\d{6}Z"` to extract it from the batch file name)
2. Use `striptime` to check if its a real/possible datetime?

Reply:
Yeah, exactly! If we keep making the regex more and more complex, we’ll end up reimplementing strptime.

Re your new plan, I solemnly give my blessing 👼